File: day_1/part_2.py
# first attempt
input = open('puzzle_input', 'r')
elves = []
elf = 0

for line in input:
    if line == '\n':
        if elf != 0:
            elves.append(elf)
        elf = 0
    else:
        elf += int(line)

elves.sort(reverse=True)
print(sum(elves[:3]))

# one liner
print(sum(sorted([sum(list(map(int, i.split('\n')))) for i in open('puzzle_input', 'r').read().strip('\n').split("\n\n")])[-3:]))

File: day_1/part_2.rs
use std::fs;

fn main() {
    let input = fs::read_to_string("puzzle_input")
        .expect("This should be able to read in the file");

    let split = input.split("\n");
    let split: Vec<&str> = split.collect();

    let mut elves: Vec<i32> = Vec::new();
    let mut elf: i32 = 0;
    for i in &split {
        let line: &str = i;
        match line.parse::<i32>() {
            Ok(n) => elf += n,
            Err(_e) => {
                if elf != 0 {
                    elves.push(elf);
                    elf = 0;
                }
            }
        }
    }

    elves.sort();
    let max_three = elves.split_off(elves.len() - 3);
    let sum: i32 = max_three.iter().sum();
    println!("{sum}");
}
